### PR TITLE
Feature(as-types): implemented Serializable on exported Objects

### LINKED ISSUE #211: Remove Args methods and extend from Serializable

### PR DESCRIPTION
**Pull request by 0xSwapFeeder**
Aim's to resolve #211 from meta #205.

Updated tests as well.

Therefore **Breaking Changes**!

### CODE REVIEW

**Review by Ben-Rey — APPROVED** [review]

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: packages/as-types/assembly/__tests__/dto-tests/Person.ts [hunk 1]
```diff
@@ -13,27 +14,38 @@ export class Divinity extends Person implements Serializable {
...
-    this.name = args.nextString().expect("Can't deserialize the name");
+    const resultAge = args.nextI32();
+    const resultName = args.nextString();
+
+    if (resultAge.isErr()) {
+      return new Result(0, "Can't deserialize Age.");
+    }
+    if (resultName.isErr()) {
+      return new Result(0, "Can't deserialize Name.");
+    }
```
Comment (gregLibert) [inline packages/as-types/assembly/__tests__/dto-tests/Person.ts:25]:
```suggestion
    const resultAge = args.nextI32();
    if (resultAge.isErr()) {
      return new Result(0, "Can't deserialize Age.");
    }
    
    const resultName = args.nextString();
    if (resultName.isErr()) {
      return new Result(0, "Can't deserialize Name.");
    }
```

File: packages/as-types/assembly/__tests__/dto-tests/Person.ts [hunk 2]
```diff
@@ -13,27 +14,38 @@ export class Divinity extends Person implements Serializable {
...
+    if (resultAge.isErr()) {
+      return new Result(0, "Can't deserialize Age.");
     }
-    this.age = age.unwrap();
-
-    const name = args.nextString();
-    if (name.isErr()) {
-      return new Result(0, "Can't deserialize the name");
+    if (resultName.isErr()) {
+      return new Result(0, "Can't deserialize Name.");
```
Comment (gregLibert) [inline packages/as-types/assembly/__tests__/dto-tests/Person.ts:44]:
```suggestion
    const resultAge = args.nextI32();
    if (resultAge.isErr()) {
        return new Result(0, "Can't deserialize Age.");
    }
    
    const resultName = args.nextString();
    if (resultName.isErr()) {
      return new Result(0, "Can't deserialize Name.");
    }
```

File: packages/as-types/assembly/amount.ts [hunk 3]
```diff
@@ -60,10 +60,19 @@ export class Amount implements Serializable {
    */
   deserialize(data: StaticArray<u8>, offset: i32): Result<i32> {
     let args = new Args(data, offset);
-    this.value = args.nextU64().expect("Can't deserialize the Amount value");
-    this.currency = args
-      .nextSerializable<Currency>()
-      .expect("Can't deserialize the Amount currency") as Currency;
+    const resultValue = args.nextU64();
```
Comment (gregLibert) [inline packages/as-types/assembly/amount.ts:63]: same

File: packages/as-types/assembly/currency.ts [hunk 4]
```diff
@@ -54,10 +54,19 @@ export class Currency implements Serializable {
    */
   deserialize(data: StaticArray<u8>, offset: i32): Result<i32> {
     let args = new Args(data, offset);
-    this.name = args.nextString().expect("Can't deserialize the Currency name");
-    this.minorUnit = args
-      .nextU8()
-      .expect("Can't deserialize the Currency minor unit");
+    const resultName = args.nextString();
```
Comment (gregLibert) [inline packages/as-types/assembly/currency.ts:57]: same

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: packages/as-types/assembly/__tests__/dto-tests/Person.ts (hunk 1 above)
Comment (gregLibert) [reply packages/as-types/assembly/__tests__/dto-tests/Person.ts:25]: Still not fix on my side

**Review by gregLibert — CHANGES_REQUESTED** [review]

File: packages/as-types/assembly/__tests__/dto-tests/Person.ts (hunk 1 above)
Comment (Ben-Rey) [reply packages/as-types/assembly/__tests__/dto-tests/Person.ts]: My mistake, done

**Pull request merged**